File: Pin_Attack.py
import string

def find_pin(pin):
    for i in range(0, 9999):
        if i == pin:
            return i
        else:
            print(i)

def find_password(pin):
# test every combination of letters until we find the password
    letters = list(string.printable)
    for i in letters:
        for j in letters:
            for k in letters:
                password = i + j + k 
                if password == pin:
                    return password
                else:
                    print(password)

def main():
    #  pin can be only length of 4 characters
    pin = input("Enter a 3 digit pin: ")
    if len(pin) == 3:
        print("your password is ",find_password(pin))
    else :
        print("Invalid pin")
        main()

if __name__ == "__main__":
    main()

File: Pin_Attack.c
#include <string.h>
#include <stdio.h>

int find_password(pin){
// test every combination of letters until we find the password
// create a list holding ascii characters using string header
    char letters[] = "01234566789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
    for (int i = 0; i < strlen(letters); i++){
        for (int j = 0; j < strlen(letters); j++){
            for (int k = 0; k < strlen(letters); k++){
                // put the letters together to form a password
                char password[4];
                password[0] = letters[i];
                password[1] = letters[j];
                password[2] = letters[k];
                password[3] = '\0';
                // test the password = to pin
                if (pin == hash(password)){
                    printf("%s", password);
                    return 0;
                }
                else:
                    print(password)}}}
}

void main(){
    char pin[] = scanf("Enter a 3 digit pin: ");
    // if pin is size 3 then run find_password
    if (strlen(pin) == 3){
        printf("your password is ",find_password(pin));}
    else{
        printf("Pin must be 3 digits");}
}